packages/openapi-ts/src/utils/write/types.ts

```diff
@@ -184,6 +186,116 @@ const processType = (client: Client, model: Model, onNode: OnNode) => {
+    if (model.meta?.hasTransformer) {
+      // Transform already created (maybe the model is used in other models) so we just bail here
+      return;
+    }
+
+    const generateForProperty = (
+      rootPath: string[],
+      property: Model,
+      isArrayElement?: boolean,
+    ) => {
+      const localPath = [...rootPath, property.name];
+
+      if (
+        property.type === 'string' &&
+        property.export !== 'array' &&
+        (property.format === 'date-time' || property.format === 'date')
+      ) {
+        return [
+          compiler.transform.dateTransformMutation({
+            path: localPath,
+          }),
+        ];
+      } else if (isArrayElement) {
+        // Avoid infinite recursion on arrays
+        return [];
+      } else {
+        // otherwise we recurse in case it's an object/array, and if it's not that will just bail with []
+        return generateForModel(localPath, property);
+      }
+    };
+
+    function generateForArray(localPath: string[], localModel: Model) {
+      if (localModel.export !== 'array') {
+        throw new Error(
+          'generateForArray should only be called with array models',
+        );
+      }
+
+      if (localModel.$refs.length === 1) {
+        const nextModel = client.models.find(
+          (m) => m.meta!.name === localModel.type,
+        );
+        if (!nextModel) {
+          throw new Error(
+            `Model ${localModel.type} could not be founded when building array transform`,
+          );
+        }
+
+        generateTransform(client, nextModel, onNode);
+
+        return [
+          compiler.transform.arrayTransformMutation({
+            path: localPath,
+            transformer: nextModel.meta!.name,
+          }),
+        ];
+      }
+
+      if (localModel.format === 'date' || localModel.format === 'date-time') {
+        return [
+          compiler.transform.mapArray({
+            path: localPath,
+            transformExpression: compiler.transform.newDate({
+              parameterName: 'item',
+            }),
+          }),
+        ];
+      }
+
+      // Not transform for this type
+      return [];
+    }
+
+    function generateForModel(
+      localPath: string[],
+      localModel: Model,
+    ): ts.Statement[] {
+      switch (localModel.export) {
+        case 'interface':
+          return localModel.properties.flatMap((property) =>
+            generateForProperty(localPath, property),
+          );
+        case 'array':
+          return generateForArray(localPath, localModel);
+
+        default:
+          // Unsupported
+          return [];
+      }
+    }
+
+    const transformStatements = generateForModel(['data'], model);
+    if (transformStatements.length > 0) {
+      const transformFunction = compiler.transform.transformMutationFunction({
+        modelName: model.name,
+        statements: transformStatements,
+      });
+
+      client.types[model.meta!.name].hasTransformer = true;
```

Comment:
By setting this all the later codegen can determine whether to emit a responseTransformer and concrete import